### DISCUSSION TITLE
要不直接把二维码在命令行渲染出来，免得再打开文件麻烦

### DISCUSSION THREAD

**Discussion by nnnewb** [takayama-lily/oicq#192] [Ideas]
目前想到的方案其实就两种，一种把图片按像素解析了再渲染到命令行，例如 [terminal-image](https://www.npmjs.com/package/terminal-image) 这个库做的事情。

```javascript
import terminalImage from 'terminal-image';

client.on('system.login.qrcode', async (data) => {
  console.log(await terminalImage.buffer(data.image))
})
```

![image](https://user-images.githubusercontent.com/16402753/128308301-35ca99c5-58cb-4d67-9a73-f018300dd668.png)

或者把二维码内容解析了，再重新编码渲染，有点脱裤子放屁的感觉。不过我试了下效果比 terminal-image 好。

```ts
import { createClient } from 'oicq';
import { config } from 'dotenv';
import jsQr from 'jsqr';
import Jimp from 'jimp';
import t from 'qrcode-terminal';

// load dotenv
config();

// login
const client = createClient(parseInt(process.env.MIRAI_WEBHOOKS_UIN || ''), { platform: 5 });
client.login();

client.on('system.login.qrcode', (data) => {
  Jimp.read(data.image).then((img) => {
    const qrcode = jsQr(new Uint8ClampedArray(img.bitmap.data), img.bitmap.width, img.bitmap.height);
    if (qrcode) {
      t.generate(qrcode.data, { small: true }, (str) => console.log(str));
    }
  });
});
```

![image](https://user-images.githubusercontent.com/16402753/128308469-b78c6e54-d0e9-4c83-83cb-4acefaef9a9d.png)

**Comment by takayama-lily** [comment]
之前想这么弄，引入了qrcode这个包，但这包要2.5M比其他加起来都多，就为打印一这个

**Reply by takayama-lily** [reply]
另外，qrcode似乎无法正常打印，jimp依赖包体积过大，不知道还有没有什么可用的

**Reply by nnnewb** [reply]
[pngjs](https://www.npmjs.com/package/pngjs)
这个怎么样？还没动手用过， 不过 Unpacked Size 只有 600kb+，也没有其他依赖。

**Reply by nnnewb** [reply]
算了下，jsqr、qrcode-terminal、pngjs 合起来大约要 1M 左右（算npmjs上的unpacked size）

**Reply by takayama-lily** [reply]
可以用（贴贴！
